lib/std/unicode.zig

```diff
@@ -293,68 +341,82 @@ pub const Utf8Iterator = struct {
     }
 };
 
-pub const Utf16LeIterator = struct {
-    bytes: []const u8,
-    i: usize,
+pub fn Utf16Iterator(comptime endianness: std.builtin.Endian) type {
+    return struct {
+        bytes: []const u8,
+        i: usize,
 
-    pub fn init(s: []const u16) Utf16LeIterator {
-        return Utf16LeIterator{
-            .bytes = mem.sliceAsBytes(s),
-            .i = 0,
-        };
-    }
+        pub fn init(s: []const u16) @This() {
+            return .{
+                .bytes = mem.sliceAsBytes(s),
+                .i = 0,
+            };
+        }
 
-    pub fn nextCodepoint(it: *Utf16LeIterator) !?u21 {
-        assert(it.i <= it.bytes.len);
-        if (it.i == it.bytes.len) return null;
-        const c0: u21 = mem.readIntLittle(u16, it.bytes[it.i..][0..2]);
-        it.i += 2;
-        if (c0 & ~@as(u21, 0x03ff) == 0xd800) {
-            // surrogate pair
-            if (it.i >= it.bytes.len) return error.DanglingSurrogateHalf;
-            const c1: u21 = mem.readIntLittle(u16, it.bytes[it.i..][0..2]);
-            if (c1 & ~@as(u21, 0x03ff) != 0xdc00) return error.ExpectedSecondSurrogateHalf;
+        pub fn nextCodepoint(it: *@This()) !?u21 {
+            assert(it.i <= it.bytes.len);
+            if (it.i == it.bytes.len) return null;
+            const c0: u21 = switch (endianness) {
+                .Little => mem.readIntLittle(u16, it.bytes[it.i..][0..2]),
```

Comment:
Instead of switching on `endianness`, why not use `mem.readInt` like so:

```zig
const c0 = mem.readInt(u16, it.bytes[it.i..][0..2], endianness);
```

Reply:
Updated everywhere.